lib/web-ledger-client.js

```diff
@@ -0,0 +1,215 @@
+/*!
+ * Copyright (c) 2018 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const url = require('url');
+const axios = require('axios');
+const https = require('https');
+
+const DEFAULT_MODE = 'test';
+const DEFAULT_HEADERS = {
+  'accept': 'application/ld+json, application/json'
+};
+
+const WEB_LEDGER_CONTEXT_V1 = 'https://w3id.org/webledger/v1';
+
+class WebLedgerClient {
+  /**
+   * @param injector {Injector}
+   *
+   * @param hostname {string} Hostname of the ledger (points to a load balancer
+   *   or a specific node).
+
+   * @param [mode='test'] {string} One of 'dev'/'test'/'live'. Determines https
+   *   agent settings (does not reject unsigned certs in 'dev' mode, for
+   *   example).
+   * @param [logger]
+   * @param [httpsAgent] {Agent}
+   */
+  constructor({injector, hostname, mode, logger, httpsAgent}) {
+    this.injector = injector;
+
+    this.hostname = hostname;
+    if(!hostname) {
+      throw new Error('Missing ledger hostname');
+    }
+
+    this.mode = mode || DEFAULT_MODE;
+    this.logger = logger || console;
+
+    this.httpsAgent = httpsAgent;
+    if(mode === 'dev' && !this.httpsAgent) {
+      this.httpsAgent = new https.Agent({rejectUnauthorized: false});
+    }
+  }
+
+  /**
+   * Fetches a Record for a given ID.
+   *
+   * @param id {string} ID uri
+   *
+   * @returns {Promise<object>} Resolves to Record Fetch Result
+   */
+  async get({id}) {
+    if(!id) {
+      throw new Error('Invalid or missing Record URI');
+    }
+
+    const ledgerAgent = await this.getAgent();
+    const queryServiceUrl = url.parse(ledgerAgent.service.ledgerQueryService);
+    queryServiceUrl.search = `id=${id}`;
+    const idUrl = url.format(queryServiceUrl);
+
+    this.logger.log(`Retrieving Record [ledger: ${this.hostname}]:`, id);
+
+    let response;
+    try {
+      response = await axios.post(idUrl, {}, {
+        httpsAgent: this.httpsAgent,
+        headers: DEFAULT_HEADERS
+      });
+    } catch(error) {
+      if(error.response && error.response.status !== 404) {
+        this.logger.error('Failed to fetch Record:', error);
+      }
+      throw error;
+    }
+
+    const body = response.data;
+
+    const result = {
+      found: true,
+      retry: false,
+      id,
+      hostname: this.hostname,
+      meta: body.meta
+    };
+
+    const record = body.object;
```

Comment:
In the general Web Ledger case, will the payload still reside in the `object` property?

Reply:
The payload will reside in the `record` property. Just ran into the problem and fixed it https://github.com/digitalbazaar/web-ledger-client/pull/10/commits/c80ca8868403e1af6c247fde2c239014842a375f. Most of the code has not been tested as this is still a WIP.